### PR TITLE
Support aggregation `AVG`

### PR DESCRIPTION
**Pull request by zmrdltl**
# Changes
#### Added code to test for columns with and without null, respectively.

### CODE REVIEW

File: src/executor/aggregate/state.rs [hunk 1]
```diff
@@ -90,13 +89,20 @@ impl<'a> State<'a> {
...
-            .map(|(k, (_, v))| (k, v))
+            .map(|(k, v)| (k, v))
             .chunks(size)
             .into_iter()
             .enumerate()
             .map(|(i, entries)| {
                 let aggregated = entries
-                    .map(|((_, aggr), value)| (aggr, value))
+                    .map(|((_, aggr), v)| {
+                        //v.2 : exists to store the intermediate value of the avg function.
```
Comment (ever0de) [inline src/executor/aggregate/state.rs:99]: Format comment 

```suggestion
                        // v.2 : exists to store the intermediate value of the avg function.
```
ref: <https://doc.rust-lang.org/reference/comments.html>

Comment (ever0de) [reply src/executor/aggregate/state.rs]: Resolved commit: <506f34fc51e77e060450984253e7244c76d4cf8f>

**Comment by zmrdltl** [comment]
only cargo build success. test is failed. just for the structure of mod.rs, state.rs

**Comment by zmrdltl** [comment]
Refactored mod.rs, state.rs structure with private enum AggrValue. 
Thanks for the help with systax and interfaces @panarch.

**Review by panarch — APPROVED** [review]
Awesome!!
All looks good, merging 👍 👍 👍

**Pull request merged**